### ISSUE TITLE
Distance matrix similar to PyTorch cdist.

### ISSUE THREAD

**Issue by diegogcerdas** [maxvanspengler/hyperbolic_learning_library#44]
For some loss functions (e.g. pairwise cross-entropy), it would be helpful to compute (in a batched manner) the distance between each pair from two collections of inputs `x1`, `x2`, using a specific `Manifold`.
It would be similar to PyTorch's [cdist](https://pytorch.org/docs/stable/generated/torch.cdist.html): "if `x1` has shape (B,P,M) and `x2` has shape (B,R,M), then the output will have shape (B,P,R)".

**Comment by diegogcerdas** [comment]
Closed by #47.
